shipyard_blueprints/shipyard-bp-utils/shipyard_bp_utils/files.py

```diff
@@ -289,6 +290,72 @@ def find_all_file_matches(file_names, file_name_re):
     return matching_file_names
 
 
+def find_files_by_regex_or_exact_match(
+    start_path=".", pattern=".*", exact_match=False, valid_extensions=None
+):
+    """Find files in the given directory and its subdirectories that match the given regex pattern.
+
+    Args:
+        start_path (str): The starting path for the directory search. Defaults to the current directory.
+        pattern (str): The regex pattern to match filenames against.
+        exact_match (bool): Whether to match the pattern exactly or just contain it.
+        valid_extensions (set): A set of valid file extensions to match against. ex: {".csv", ".xlsx", ".xls"}.
+
+    Returns:
+        list: A list of filenames (with their full paths) that match the given regex pattern.
+    """
+    print("=== File Search Logs ===")
```

Comment:
Added this print log styling because the print and logger statements do not log sequentially